src/server/scripts/Spells/spell_hunter.cpp

```diff
@@ -1012,9 +1013,48 @@ class spell_hun_tnt : public SpellScriptLoader
         }
 };
 
+// -186257 - Aspect of the cheetah
+class spell_hun_aspect_cheetah : public SpellScriptLoader
+{
+    public:
+        spell_hun_aspect_cheetah() : SpellScriptLoader("spell_hun_aspect_cheetah") { }
+
+        class spell_hun_aspect_cheetah_AuraScript : public AuraScript
+        {
+
+            PrepareAuraScript(spell_hun_aspect_cheetah_AuraScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                return ValidateSpellInfo
+                ({
+                    SPELL_HUNTER_ASPECT_CHEETAH_SLOW
+                });
+            }
+
+            void HandleOnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+            {
+                Unit *target = GetTarget();
+                target->CastSpell(target, SPELL_HUNTER_ASPECT_CHEETAH_SLOW, true);
```

Comment:
Currently, applies the second aura ignoring how it was removed. To be honest, I have not tested that on retail.